src/SmartComponents/AddSystemModal/AddSystemModal.js

```diff
@@ -243,10 +243,10 @@ export class AddSystemModal extends Component {
                         >
                             <SystemsTable
                                 selectedSystemIds={ selectedSystemIds }
-                                hasHistoricalDropdown={ hasHSPReadPermissions }
+                                hasHistoricalDropdown={ permissions.hspRead }
```

Comment:
Observation: `SystemsTable` already gets `permissions`.

Reply:
So, I only use `<PermissionContext.Consumer>` at the top level components because it makes testing a bit easier and my snapshots less verbose.

Regarding `SystemsTable` already having permissions. Before, we passed `hasHistoricalDropdown` always as true from here because when the `SystemsTable` is being rendered from `AddSystemModal` we always show the historical dropdown. With this, I can decide to render the dropdown depending on whether or not the user has permissions to read hsps.

Reply:
Oh, I see the nuance of the testing. With a helper method within tests it could be doable with no so much hassle. Check https://testing-library.com/docs/example-react-context/. If you find this `permissions` passing as a better option, then feel free to keep it as is. 

As for the `hasHistoricalDropdown` it makes sense as it is.